src/main/groovy/nebula/plugin/publishing/PublicationBase.groovy

```diff
@@ -0,0 +1,87 @@
+package nebula.plugin.publishing
+
+import nebula.plugin.publishing.ivy.IvyBasePublishPlugin
+import nebula.plugin.publishing.maven.MavenBasePublishPlugin
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.ivy.IvyPublication
+import org.gradle.api.publish.ivy.plugins.IvyPublishPlugin
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+
+trait PublicationBase implements Plugin<Project>, StandardTextValues {
+
+    /**
+     * Checks to see if the task already exists or not on the project.  If it does not, then it creates it.
+     * @param project project to add the task too
+     * @param task Map configuration of the task to create
+     * @return Null if it was not created, the task object if it was.
+     */
+    Task addTaskLocal(Project project, Map<String, ?> task) {
+        if (!project.tasks.findByName(task.get('name').toString())) {
+            project.tasks.create(task)
+        } else {
+            null
+        }
+    }
+
+    void checkAddMavenPublish(Project project, Task createdTask) {
+        project.plugins.withType(MavenPublishPlugin) {
+            project.plugins.apply(MavenBasePublishPlugin)
+
+            project.publishing.publications {
```

Comment:
wow, thats going to be tough because its buried about three layers of custom plugins deep in a CI server.  It has to do with the core publish being in the core plugin that everything else inherits, with child plugins and such.  Its really tough, I do know its a well known groovy/gradle issue because it only took me a minute or two to find the solution when i hit it.  I can track down the research when im on my work machine tomorrow.